File: 98. 验证二叉搜索树.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    
    public boolean isValid=true;
    public TreeNode pre=null;
    
    public boolean isValidBST(TreeNode root) {
        
        inOrder(root);
        return isValid;
    }
   
    private void inOrder(TreeNode root){
        if(root!=null){
        
            inOrder(root.left);
            if(pre!=null){
                
                if(root.val<=pre.val)
                    isValid=false;
            }
            pre=root;
            inOrder(root.right);
        }
        
    }
}

File: js/98. 验证二叉搜索树.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isValidBST = function(root) {
    if(!root)
        return null
    let isValid=true
    let pre
    const traverse=function(root){
        root.left&&traverse(root.left)

        if(!pre)
            pre=root
        else{
            if(pre.val>=root.val)
                isValid=false
            pre=root
        }
        root.right&&traverse(root.right)
    }
    traverse(root)
    return isValid
};
